src/sentry/receivers/onboarding.py

```diff
@@ -165,7 +166,10 @@ def record_release_received(project, group, event, **kwargs):
 
 @event_processed.connect(weak=False)
 def record_user_context_received(project, group, event, **kwargs):
-    if not event.data.get('sentry.interfaces.User'):
+    user_context = event.data.get('sentry.interfaces.User')
+    if not user_context:
+        return
+    elif isinstance(user_context, dict) and iterkeys(user_context) != ['ip_address']:
```

Comment:
Remove this `dict` check and the `iterkeys` since this won't work. Also remove it in the other place where you're doing the same check.